uarray/_uarray_dispatch.cxx

```diff
@@ -498,18 +498,36 @@ LoopReturn for_each_backend(const std::string & domain_key, Callback call)
   auto should_skip =
     [&](PyObject * backend)
       {
+        bool success = true;
         auto it = std::find_if(
           skip.begin(), skip.end(),
-          [&](const py_ref & be) { return be.get() == backend; });
+          [&](const py_ref & be)
+          {
+            int result = PyObject_RichCompareBool(be.get(), backend, Py_EQ);
+            if (result < 0)
+            {
+              success = false;
+              return 0;
+            }
+            return result;
```

Comment:
How about
```cpp
auto result = PyObject_RichCompareBool(be.get(), backend, Py_EQ);
success = (result >= 0);
return (result != 0);
```
Not only is it a bit more compact, it also breaks out of the `find_if` on error.